<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>TASSELpy.net.maizegenetics.dna.snp package &mdash; TASSELpy 0.1.dev-20140723133829 documentation</title>
    
    <link rel="stylesheet" href="_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.dev-20140723133829',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="TASSELpy 0.1.dev-20140723133829 documentation" href="index.html" />
    <link rel="up" title="TASSELpy.net.maizegenetics.dna package" href="TASSELpy.net.maizegenetics.dna.html" />
    <link rel="next" title="TASSELpy.net.maizegenetics.dna.snp.depth package" href="TASSELpy.net.maizegenetics.dna.snp.depth.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="TASSELpy.net.maizegenetics.dna.snp.depth.html" title="TASSELpy.net.maizegenetics.dna.snp.depth package"
             accesskey="N">next</a> |</li>
        <li><a href="index.html">TASSELpy 0.1.dev-20140723133829 documentation</a> &raquo;</li>
          <li><a href="TASSELpy.html" >TASSELpy package</a> &raquo;</li>
          <li><a href="TASSELpy.net.html" >TASSELpy.net package</a> &raquo;</li>
          <li><a href="TASSELpy.net.maizegenetics.html" >TASSELpy.net.maizegenetics package</a> &raquo;</li>
          <li><a href="TASSELpy.net.maizegenetics.dna.html" accesskey="U">TASSELpy.net.maizegenetics.dna package</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="tasselpy-net-maizegenetics-dna-snp-package">
<h1>TASSELpy.net.maizegenetics.dna.snp package<a class="headerlink" href="#tasselpy-net-maizegenetics-dna-snp-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="subpackages">
<h2>Subpackages<a class="headerlink" href="#subpackages" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.depth.html">TASSELpy.net.maizegenetics.dna.snp.depth package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.depth.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.depth.html#module-TASSELpy.net.maizegenetics.dna.snp.depth.AlleleDepth">TASSELpy.net.maizegenetics.dna.snp.depth.AlleleDepth module</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.depth.html#module-TASSELpy.net.maizegenetics.dna.snp.depth">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.score.html">TASSELpy.net.maizegenetics.dna.snp.score package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.score.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.score.html#module-TASSELpy.net.maizegenetics.dna.snp.score.SiteScore">TASSELpy.net.maizegenetics.dna.snp.score.SiteScore module</a><ul class="simple">
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="TASSELpy.net.maizegenetics.dna.snp.score.html#module-TASSELpy.net.maizegenetics.dna.snp.score">Module contents</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable">
<span id="tasselpy-net-maizegenetics-dna-snp-coregenotypetable-module"></span><h2>TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable module<a class="headerlink" href="#module-TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable.CoreGenotypeTable">
<em class="property">class </em><tt class="descclassname">TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable.</tt><tt class="descname">CoreGenotypeTable</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/CoreGenotypeTable.html#CoreGenotypeTable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable.CoreGenotypeTable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable"><tt class="xref py py-class docutils literal"><span class="pre">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable</span></tt></a></p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">alleleDefinitions</span></tt>(*args)</td>
<td>For the first signature: Returns allele values as strings for all sites.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllSites</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each site matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllTaxa</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether site at</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForSitesBlock</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">alleles</span></tt>(*args)</td>
<td>Returns all alleles at given site in order of frequency.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelesBySortType</span></tt>(*args)</td>
<td>Returns all alleles at given site in order defined by scope</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelesSortedByFrequency</span></tt>(*args)</td>
<td>Return sorted list of alleles from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">bitStorage</span></tt>(*args)</td>
<td>Returns BitStorage for this Genotype</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomalPosition</span></tt>(*args)</td>
<td>Returns the physical position at given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosome</span></tt>(*args)</td>
<td>Returns Chromosome for given site or with matching name</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeName</span></tt>(*args)</td>
<td>Return chromosome name for given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeSiteCount</span></tt>(*args)</td>
<td>Return number of sites for given chromosome</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomes</span></tt>(*args)</td>
<td>Return all chromosomes</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomesOffsets</span></tt>(*args)</td>
<td>Returns starting site for each chromosome</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">clone</span></tt>(*args)</td>
<td>Creates and returns a copy of this object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">compositeAlignments</span></tt>(*args)</td>
<td>Returns individual genotype tables within this genotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">depth</span></tt>(*args)</td>
<td>Returns allele depth object (null if not present)</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">depthForAlleles</span></tt>(*args)</td>
<td>Returns depth count for each diploid allele at the given taxon and site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">diploidAsString</span></tt>(*args)</td>
<td>Returns String representation of diploid allele value at site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">equals</span></tt>(*args)</td>
<td>Indicates whether some other object is &#8220;equal to&#8221; this one</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">firstLastSiteOfChromosome</span></tt>(*args)</td>
<td>Get the first (inclusive) and last (exclusive) site of the specified</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genoCounts</span></tt>(*args)</td>
<td>Returns counts of all diploid combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genomeVersion</span></tt>(*args)</td>
<td>Gets the Genome Assembly</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genosSortedByFrequency</span></tt>(*args)</td>
<td>Returns sorted list of diploid values from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotype</span></tt>(*args)</td>
<td>Returns diploid value (genotype) for a given taxon and site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllSites</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllTaxa</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeArray</span></tt>(*args)</td>
<td>Returns diploid values for given taxon and site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsString</span></tt>(*args)</td>
<td>Returns string representation of diploid values returned by genotype() for given taxon and site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringArray</span></tt>(*args)</td>
<td>Returns string representation of diploid values returned by genotypeArray() for given taxon and site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRange</span></tt>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRow</span></tt>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon for all sites.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeMatrix</span></tt>(*args)</td>
<td>Returns the immutable genotype matrix.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeRange</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">getArray</span></tt>(size)</td>
<td>Gets an empty wrapped java array that can accept the type of the wrapped</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">getDblArray</span></tt>(rows[,&nbsp;cols])</td>
<td>Gets an empty wrapped java array that can accept the type of other wrapped java arrays: i.e.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllSites</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllTaxa</span></tt>(*args)</td>
<td>Returns a sequence of true/false values indicating whether site at each taxon for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForSitesBlock</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at sites (in given blocks, 64 sites per block including start block but excluding end block) for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasDepth</span></tt>(*args)</td>
<td>Returns true if this genotype table has sequencing depth</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasReference</span></tt>(*args)</td>
<td>Returns whether this genotype table has defined reference sequence.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasSiteScores</span></tt>(*args)</td>
<td>Returns true if this genotype table has site scores</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hashCode</span></tt>(*args)</td>
<td>Returns a hash code vlaue for the object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCount</span></tt>(*args)</td>
<td>Returns number of heterozygous taxa at given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCountForTaxon</span></tt>(*args)</td>
<td>Returns number of heterozygous sites at given taxon</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">indelSize</span></tt>(*args)</td>
<td>Returns size of indel at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isAllPolymorphic</span></tt>(*args)</td>
<td>Returns whether all sites are polymorphic</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isHeterozygous</span></tt>(*args)</td>
<td>Returns whether allele values at given taxon and site are heterozygous.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isIndel</span></tt>(*args)</td>
<td>Returns whether given site is an indel</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPhased</span></tt>(*args)</td>
<td>Returns whether this genotype table is phased</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPolymorphic</span></tt>(*args)</td>
<td>Returns whether given site is polymorphic</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPositiveStrand</span></tt>(*args)</td>
<td>Returns whether is positive strand at given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAllele</span></tt>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleAsString</span></tt>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleCount</span></tt>(*args)</td>
<td>Returns the major allele count for given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleFrequency</span></tt>(*args)</td>
<td>Returns frequency for major allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorMinorCounts</span></tt>(*args)</td>
<td>Returns counts of all major/minor allele combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">maxNumAlleles</span></tt>(*args)</td>
<td>Returns max number of alleles defined for any given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAllele</span></tt>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleAsString</span></tt>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleCount</span></tt>(*args)</td>
<td>Returns the minor allele count for given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleFrequency</span></tt>(*args)</td>
<td>Returns frequency for most common minor allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleles</span></tt>(*args)</td>
<td>Returns all minor alleles at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">numChromosomes</span></tt>(*args)</td>
<td>Return number of chromosomes</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">numberOfSites</span></tt>(*args)</td>
<td>Returns total number of sites of this genotype table</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">numberOfTaxa</span></tt>(*args)</td>
<td>Returns the number of taxa</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">physicalPositions</span></tt>(*args)</td>
<td>Returns all physical positions</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">positions</span></tt>(*args)</td>
<td>Return the position list for the gneotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAllele</span></tt>(*args)</td>
<td>Returns reference diploid allele values at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleleForAllSites</span></tt>(*args)</td>
<td>Returns reference sequence of diploid allele values.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleles</span></tt>(*args)</td>
<td>Returns reference sequence of diploid allele values for given taxon in specified range (end site not included).</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">retainsRareAlleles</span></tt>(*args)</td>
<td>Returns true if this genotype table retains rare alleles.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteName</span></tt>(*args)</td>
<td>Get SNP ID for specified site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteOfPhysicalPosition</span></tt>(*args)</td>
<td>Returns sites of given physical position/SNP ID in chromosome.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScore</span></tt>(*args)</td>
<td>Returns the site score of the given taxon and site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScoreType</span></tt>(*args)</td>
<td>Return what type of site scores this genotype table has</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScores</span></tt>(*args)</td>
<td>Returns the site scores</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">taxa</span></tt>(*args)</td>
<td>Return taxa list of this genotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">taxaName</span></tt>(*args)</td>
<td>Returns taxa name at given index</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">toString</span></tt>(*args)</td>
<td>Returns a string representation of the object</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForSite</span></tt>(*args)</td>
<td>Returns total number of non-missing allele values for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForTaxon</span></tt>(*args)</td>
<td>Returns total number of non-missing allele values for given taxon.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForSite</span></tt>(*args)</td>
<td>Returns total number of non-missing taxa for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForTaxon</span></tt>(*args)</td>
<td>Returns total number of non-missing sites for given taxon.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">wrap_existing_array</span></tt>(arr_instance)</td>
<td>Wraps a java array of this class&#8217;s type</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable.CoreGenotypeTable.__init__">
<tt class="descname">__init__</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/CoreGenotypeTable.html#CoreGenotypeTable.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable.CoreGenotypeTable.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Instantiates a CoreGenotypeTable</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>CoreGenotypeTable</em> (GenotypeCallTable <strong>genotype</strong>, PositionList <strong>positionList</strong>, TaxaList <strong>taxaList</strong>, SiteScore <strong>siteScore</strong>, AlleleDepth <strong>alleleDepth</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><em>CoreGenotypeTable</em> (GenotypeCallTable <strong>genotype</strong>, PositionList <strong>positionList</strong>, TaxaList <strong>taxaList</strong>)</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable">
<span id="tasselpy-net-maizegenetics-dna-snp-filtergenotypetable-module"></span><h2>TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable module<a class="headerlink" href="#module-TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable">
<em class="property">class </em><tt class="descclassname">TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.</tt><tt class="descname">FilterGenotypeTable</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable"><tt class="xref py py-class docutils literal"><span class="pre">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable</span></tt></a></p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">alleleDefinitions</span></tt>(*args)</td>
<td>For the first signature: Returns allele values as strings for all sites.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllSites</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each site matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllTaxa</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether site at</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForSitesBlock</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">alleles</span></tt>(*args)</td>
<td>Returns all alleles at given site in order of frequency.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelesBySortType</span></tt>(*args)</td>
<td>Returns all alleles at given site in order defined by scope</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">allelesSortedByFrequency</span></tt>(*args)</td>
<td>Return sorted list of alleles from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">bitStorage</span></tt>(*args)</td>
<td>Returns BitStorage for this Genotype</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomalPosition</span></tt>(*args)</td>
<td>Returns the physical position at given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosome</span></tt>(*args)</td>
<td>Returns Chromosome for given site or with matching name</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeName</span></tt>(*args)</td>
<td>Return chromosome name for given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeSiteCount</span></tt>(*args)</td>
<td>Return number of sites for given chromosome</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomes</span></tt>(*args)</td>
<td>Return all chromosomes</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">chromosomesOffsets</span></tt>(*args)</td>
<td>Returns starting site for each chromosome</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">clone</span></tt>(*args)</td>
<td>Creates and returns a copy of this object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">compositeAlignments</span></tt>(*args)</td>
<td>Returns individual genotype tables within this genotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">depth</span></tt>(*args)</td>
<td>Returns allele depth object (null if not present)</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">depthForAlleles</span></tt>(*args)</td>
<td>Returns depth count for each diploid allele at the given taxon and site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">diploidAsString</span></tt>(*args)</td>
<td>Returns String representation of diploid allele value at site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">equals</span></tt>(*args)</td>
<td>Indicates whether some other object is &#8220;equal to&#8221; this one</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">firstLastSiteOfChromosome</span></tt>(*args)</td>
<td>Get the first (inclusive) and last (exclusive) site of the specified</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.functionalFilters" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.functionalFilters"><tt class="xref py py-obj docutils literal"><span class="pre">functionalFilters</span></tt></a>(genoTable[,&nbsp;sitesFilter,&nbsp;...])</td>
<td></td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genoCounts</span></tt>(*args)</td>
<td>Returns counts of all diploid combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genomeVersion</span></tt>(*args)</td>
<td>Gets the Genome Assembly</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genosSortedByFrequency</span></tt>(*args)</td>
<td>Returns sorted list of diploid values from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotype</span></tt>(*args)</td>
<td>Returns diploid value (genotype) for a given taxon and site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllSites</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllTaxa</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeArray</span></tt>(*args)</td>
<td>Returns diploid values for given taxon and site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsString</span></tt>(*args)</td>
<td>Returns string representation of diploid values returned by genotype() for given taxon and site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringArray</span></tt>(*args)</td>
<td>Returns string representation of diploid values returned by genotypeArray() for given taxon and site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRange</span></tt>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRow</span></tt>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon for all sites.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeMatrix</span></tt>(*args)</td>
<td>Returns the immutable genotype matrix.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">genotypeRange</span></tt>(*args)</td>
<td>Returns sequence of diploid allele values for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">getArray</span></tt>(size)</td>
<td>Gets an empty wrapped java array that can accept the type of the wrapped</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseAlignment" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseAlignment"><tt class="xref py py-obj docutils literal"><span class="pre">getBaseAlignment</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>getBaseAlignment</em> ()</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseSitesShown" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseSitesShown"><tt class="xref py py-obj docutils literal"><span class="pre">getBaseSitesShown</span></tt></a>(*args)</td>
<td>Returns sites from original alignment that are viewable (not filtered) by</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">getDblArray</span></tt>(rows[,&nbsp;cols])</td>
<td>Gets an empty wrapped java array that can accept the type of other wrapped java arrays: i.e.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstance" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstance"><tt class="xref py py-obj docutils literal"><span class="pre">getInstance</span></tt></a>(*args)</td>
<td>This returns FilterGenotypeTable with only specified subTaxaList.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveIDs" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveIDs"><tt class="xref py py-obj docutils literal"><span class="pre">getInstanceRemoveIDs</span></tt></a>(*args)</td>
<td>Removes specific IDs</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveSiteNames" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveSiteNames"><tt class="xref py py-obj docutils literal"><span class="pre">getInstanceRemoveSiteNames</span></tt></a>(*args)</td>
<td>Removes specific named sites</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllSites</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllTaxa</span></tt>(*args)</td>
<td>Returns a sequence of true/false values indicating whether site at each taxon for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForSitesBlock</span></tt>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at sites (in given blocks, 64 sites per block including start block but excluding end block) for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasDepth</span></tt>(*args)</td>
<td>Returns true if this genotype table has sequencing depth</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasReference</span></tt>(*args)</td>
<td>Returns whether this genotype table has defined reference sequence.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">hasSiteScores</span></tt>(*args)</td>
<td>Returns true if this genotype table has site scores</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hashCode</span></tt>(*args)</td>
<td>Returns a hash code vlaue for the object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCount</span></tt>(*args)</td>
<td>Returns number of heterozygous taxa at given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCountForTaxon</span></tt>(*args)</td>
<td>Returns number of heterozygous sites at given taxon</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">indelSize</span></tt>(*args)</td>
<td>Returns size of indel at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isAllPolymorphic</span></tt>(*args)</td>
<td>Returns whether all sites are polymorphic</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isHeterozygous</span></tt>(*args)</td>
<td>Returns whether allele values at given taxon and site are heterozygous.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isIndel</span></tt>(*args)</td>
<td>Returns whether given site is an indel</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPhased</span></tt>(*args)</td>
<td>Returns whether this genotype table is phased</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPolymorphic</span></tt>(*args)</td>
<td>Returns whether given site is polymorphic</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">isPositiveStrand</span></tt>(*args)</td>
<td>Returns whether is positive strand at given site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilter" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilter"><tt class="xref py py-obj docutils literal"><span class="pre">isSiteFilter</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>isSiteFilter</em> ()</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilterByRange" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilterByRange"><tt class="xref py py-obj docutils literal"><span class="pre">isSiteFilterByRange</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>isSiteFilterByRange</em> ()</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isTaxaFilter" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isTaxaFilter"><tt class="xref py py-obj docutils literal"><span class="pre">isTaxaFilter</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>isTaxaFilter</em> ()</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAllele</span></tt>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleAsString</span></tt>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleCount</span></tt>(*args)</td>
<td>Returns the major allele count for given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleFrequency</span></tt>(*args)</td>
<td>Returns frequency for major allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">majorMinorCounts</span></tt>(*args)</td>
<td>Returns counts of all major/minor allele combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">maxNumAlleles</span></tt>(*args)</td>
<td>Returns max number of alleles defined for any given site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAllele</span></tt>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleAsString</span></tt>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleCount</span></tt>(*args)</td>
<td>Returns the minor allele count for given site</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleFrequency</span></tt>(*args)</td>
<td>Returns frequency for most common minor allele at given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleles</span></tt>(*args)</td>
<td>Returns all minor alleles at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">numChromosomes</span></tt>(*args)</td>
<td>Return number of chromosomes</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">numberOfSites</span></tt>(*args)</td>
<td>Returns total number of sites of this genotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">numberOfTaxa</span></tt>(*args)</td>
<td>Returns the number of taxa</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">physicalPositions</span></tt>(*args)</td>
<td>Returns all physical positions</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">positions</span></tt>(*args)</td>
<td>Return the position list for the gneotype table</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAllele</span></tt>(*args)</td>
<td>Returns reference diploid allele values at given site.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleleForAllSites</span></tt>(*args)</td>
<td>Returns reference sequence of diploid allele values.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleles</span></tt>(*args)</td>
<td>Returns reference sequence of diploid allele values for given taxon in specified range (end site not included).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">retainsRareAlleles</span></tt>(*args)</td>
<td>Returns true if this genotype table retains rare alleles.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.reverseTranslateSite" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.reverseTranslateSite"><tt class="xref py py-obj docutils literal"><span class="pre">reverseTranslateSite</span></tt></a>(*args)</td>
<td>Returns site of this FilterGenotypeTable based on given site from</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteName</span></tt>(*args)</td>
<td>Get SNP ID for specified site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteOfPhysicalPosition</span></tt>(*args)</td>
<td>Returns sites of given physical position/SNP ID in chromosome.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScore</span></tt>(*args)</td>
<td>Returns the site score of the given taxon and site</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScoreType</span></tt>(*args)</td>
<td>Return what type of site scores this genotype table has</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">siteScores</span></tt>(*args)</td>
<td>Returns the site scores</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">taxa</span></tt>(*args)</td>
<td>Return taxa list of this genotype table</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">taxaName</span></tt>(*args)</td>
<td>Returns taxa name at given index</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">toString</span></tt>(*args)</td>
<td>Returns a string representation of the object</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForSite</span></tt>(*args)</td>
<td>Returns total number of non-missing allele values for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForTaxon</span></tt>(*args)</td>
<td>Returns total number of non-missing allele values for given taxon.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForSite</span></tt>(*args)</td>
<td>Returns total number of non-missing taxa for given site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForTaxon</span></tt>(*args)</td>
<td>Returns total number of non-missing sites for given taxon.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateSite" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateSite"><tt class="xref py py-obj docutils literal"><span class="pre">translateSite</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>translateSite</em> (int <strong>site</strong>)</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateTaxon" title="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateTaxon"><tt class="xref py py-obj docutils literal"><span class="pre">translateTaxon</span></tt></a>(*args)</td>
<td><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">signature:</th><td class="field-body"><em>translateTaxon</em> (int <strong>taxon</strong>)</td>
</tr>
</tbody>
</table>
</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">wrap_existing_array</span></tt>(arr_instance)</td>
<td>Wraps a java array of this class&#8217;s type</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.__init__">
<tt class="descname">__init__</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.__init__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="staticmethod">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.functionalFilters">
<em class="property">static </em><tt class="descname">functionalFilters</tt><big>(</big><em>genoTable</em>, <em>sitesFilter=None</em>, <em>taxaFilter=None</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.functionalFilters"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.functionalFilters" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseAlignment">
<tt class="descname">getBaseAlignment</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.getBaseAlignment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseAlignment" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>getBaseAlignment</em> ()</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseSitesShown">
<tt class="descname">getBaseSitesShown</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.getBaseSitesShown"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getBaseSitesShown" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sites from original alignment that are viewable (not filtered) by
this filter alignment</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>getBaseSitesShown</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of sites</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstance">
<em class="property">static </em><tt class="descname">getInstance</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.getInstance"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstance" title="Permalink to this definition">¶</a></dt>
<dd><p>This returns FilterGenotypeTable with only specified subTaxaList. Defaults
to retain unknown taxa. If retainUnkownTaxa is true then Alignment will return
unknown values for missing taxa</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, TaxaList <strong>subTaxaList</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, TaxaList <strong>subTaxaList</strong>, boolean <strong>retainUnknownTaxa</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, int[] <strong>subSites</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, String[] <strong>siteNamesToKeep</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, String <strong>chromosome</strong>, int <strong>startPhysicalPos</strong>, int <strong>endPhysicalPos</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, Chromosome <strong>chromosome</strong>, int <strong>startPhysicalPos</strong>, int <strong>endPhysicalPos</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, Chromosome <strong>chromosome</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstance</em> (GenotypeTable <strong>a</strong>, int <strong>startSite</strong>, int <strong>endSite</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>GenotypeTable</em>) &#8211; alignment</li>
<li><strong>subTaxaList</strong> (<em>TaxaList</em>) &#8211; subset id group</li>
<li><strong>retainUnknownTaxa</strong> (<em>boolean</em>) &#8211; whether to retain unknown taxa</li>
<li><strong>subSites</strong> (<em>int[]</em>) &#8211; subset of sites you want</li>
<li><strong>siteNamesToKeep</strong> (<em>String[]</em>) &#8211; names of the sites to keep</li>
<li><strong>chromosome</strong> (<em>Chromosome</em>) &#8211; name of chromosome</li>
<li><strong>startPhysicalPos</strong> (<em>int</em>) &#8211; starting physical position</li>
<li><strong>endPhysicalPos</strong> (<em>int</em>) &#8211; ending physical position</li>
<li><strong>startSite</strong> (<em>int</em>) &#8211; start site (inclusive)</li>
<li><strong>endSite</strong> (<em>int</em>) &#8211; end site (inclusive)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">filter alignment</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">GenotypeTable</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveIDs">
<em class="property">static </em><tt class="descname">getInstanceRemoveIDs</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.getInstanceRemoveIDs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveIDs" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes specific IDs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstanceRemoveIDs</em> (GenotypeTable <strong>a</strong>, TaxaList <strong>subTaxaList</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>GenotypeTable</em>) &#8211; alignment to filter</li>
<li><strong>subTaxaList</strong> (<em>TaxaList</em>) &#8211; specified IDs</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Filtered Alignment</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">GenotypeTable</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="staticmethod">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveSiteNames">
<em class="property">static </em><tt class="descname">getInstanceRemoveSiteNames</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.getInstanceRemoveSiteNames"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.getInstanceRemoveSiteNames" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes specific named sites</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>getInstanceRemoveSiteNames</em> (GenotypeTable <strong>a</strong>, String[] <strong>siteNamesToRemove</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>a</strong> (<em>GenotypeTable</em>) &#8211; alignment</li>
<li><strong>siteNamesToRemove</strong> (<em>String[]</em>) &#8211; names of the sites to remove</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Filtered Alignment</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">FilterGenotypeTable</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilter">
<tt class="descname">isSiteFilter</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.isSiteFilter"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilter" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isSiteFilter</em> ()</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilterByRange">
<tt class="descname">isSiteFilterByRange</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.isSiteFilterByRange"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isSiteFilterByRange" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isSiteFilterByRange</em> ()</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isTaxaFilter">
<tt class="descname">isTaxaFilter</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.isTaxaFilter"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.isTaxaFilter" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isTaxaFilter</em> ()</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.reverseTranslateSite">
<tt class="descname">reverseTranslateSite</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.reverseTranslateSite"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.reverseTranslateSite" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns site of this FilterGenotypeTable based on given site from
embedded Alignment</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>reverseTranslateSite</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">site of this FilterGenotypeTable based on given site from embedded Alignment</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateSite">
<tt class="descname">translateSite</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.translateSite"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateSite" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>translateSite</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateTaxon">
<tt class="descname">translateTaxon</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/FilterGenotypeTable.html#FilterGenotypeTable.translateTaxon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable.FilterGenotypeTable.translateTaxon" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>translateTaxon</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>taxon</strong> (<em>int</em>) &#8211; taxon</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-TASSELpy.net.maizegenetics.dna.snp.GenotypeTable">
<span id="tasselpy-net-maizegenetics-dna-snp-genotypetable-module"></span><h2>TASSELpy.net.maizegenetics.dna.snp.GenotypeTable module<a class="headerlink" href="#module-TASSELpy.net.maizegenetics.dna.snp.GenotypeTable" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable">
<em class="property">class </em><tt class="descclassname">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.</tt><tt class="descname">GenotypeTable</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.MetaGenotypeTable" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.MetaGenotypeTable"><tt class="xref py py-class docutils literal"><span class="pre">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.MetaGenotypeTable</span></tt></a></p>
<p>A representation of the SNP and indel variation for a set of taxa
and genomic positions.</p>
<p>GenotypeTable always consists of a TaxaList, PositionList, and
GenotypeCallTable. Additionally, as needed they also represent alleles
in bit form, with sequencing depth, or other scores (e.g. quality scores).</p>
<p>Use GenotypeTableBuilder to create GenotypeTable</p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleleDefinitions" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleleDefinitions"><tt class="xref py py-obj docutils literal"><span class="pre">alleleDefinitions</span></tt></a>(*args)</td>
<td>For the first signature: Returns allele values as strings for all sites.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllSites" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllSites"><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllSites</span></tt></a>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each site matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllTaxa" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllTaxa"><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForAllTaxa</span></tt></a>(*args)</td>
<td>Returns sequence of true/false values indicating whether site at</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForSitesBlock" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForSitesBlock"><tt class="xref py py-obj docutils literal"><span class="pre">allelePresenceForSitesBlock</span></tt></a>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleles"><tt class="xref py py-obj docutils literal"><span class="pre">alleles</span></tt></a>(*args)</td>
<td>Returns all alleles at given site in order of frequency.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesBySortType" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesBySortType"><tt class="xref py py-obj docutils literal"><span class="pre">allelesBySortType</span></tt></a>(*args)</td>
<td>Returns all alleles at given site in order defined by scope</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesSortedByFrequency" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesSortedByFrequency"><tt class="xref py py-obj docutils literal"><span class="pre">allelesSortedByFrequency</span></tt></a>(*args)</td>
<td>Return sorted list of alleles from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.bitStorage" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.bitStorage"><tt class="xref py py-obj docutils literal"><span class="pre">bitStorage</span></tt></a>(*args)</td>
<td>Returns BitStorage for this Genotype</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomalPosition" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomalPosition"><tt class="xref py py-obj docutils literal"><span class="pre">chromosomalPosition</span></tt></a>(*args)</td>
<td>Returns the physical position at given site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosome" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosome"><tt class="xref py py-obj docutils literal"><span class="pre">chromosome</span></tt></a>(*args)</td>
<td>Returns Chromosome for given site or with matching name</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeName" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeName"><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeName</span></tt></a>(*args)</td>
<td>Return chromosome name for given site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeSiteCount" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeSiteCount"><tt class="xref py py-obj docutils literal"><span class="pre">chromosomeSiteCount</span></tt></a>(*args)</td>
<td>Return number of sites for given chromosome</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomes" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomes"><tt class="xref py py-obj docutils literal"><span class="pre">chromosomes</span></tt></a>(*args)</td>
<td>Return all chromosomes</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomesOffsets" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomesOffsets"><tt class="xref py py-obj docutils literal"><span class="pre">chromosomesOffsets</span></tt></a>(*args)</td>
<td>Returns starting site for each chromosome</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">clone</span></tt>(*args)</td>
<td>Creates and returns a copy of this object</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.compositeAlignments" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.compositeAlignments"><tt class="xref py py-obj docutils literal"><span class="pre">compositeAlignments</span></tt></a>(*args)</td>
<td>Returns individual genotype tables within this genotype table</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depth" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depth"><tt class="xref py py-obj docutils literal"><span class="pre">depth</span></tt></a>(*args)</td>
<td>Returns allele depth object (null if not present)</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depthForAlleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depthForAlleles"><tt class="xref py py-obj docutils literal"><span class="pre">depthForAlleles</span></tt></a>(*args)</td>
<td>Returns depth count for each diploid allele at the given taxon and site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.diploidAsString" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.diploidAsString"><tt class="xref py py-obj docutils literal"><span class="pre">diploidAsString</span></tt></a>(*args)</td>
<td>Returns String representation of diploid allele value at site.</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">equals</span></tt>(*args)</td>
<td>Indicates whether some other object is &#8220;equal to&#8221; this one</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.firstLastSiteOfChromosome" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.firstLastSiteOfChromosome"><tt class="xref py py-obj docutils literal"><span class="pre">firstLastSiteOfChromosome</span></tt></a>(*args)</td>
<td>Get the first (inclusive) and last (exclusive) site of the specified</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genoCounts" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genoCounts"><tt class="xref py py-obj docutils literal"><span class="pre">genoCounts</span></tt></a>(*args)</td>
<td>Returns counts of all diploid combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genomeVersion" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genomeVersion"><tt class="xref py py-obj docutils literal"><span class="pre">genomeVersion</span></tt></a>(*args)</td>
<td>Gets the Genome Assembly</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genosSortedByFrequency" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genosSortedByFrequency"><tt class="xref py py-obj docutils literal"><span class="pre">genosSortedByFrequency</span></tt></a>(*args)</td>
<td>Returns sorted list of diploid values from highest frequency to lowest at given site in genotype table.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotype" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotype"><tt class="xref py py-obj docutils literal"><span class="pre">genotype</span></tt></a>(*args)</td>
<td>Returns diploid value (genotype) for a given taxon and site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllSites" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllSites"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllSites</span></tt></a>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllTaxa" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllTaxa"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAllTaxa</span></tt></a>(*args)</td>
<td>Returns sequence of diploid allele values for all taxa for given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeArray" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeArray"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeArray</span></tt></a>(*args)</td>
<td>Returns diploid values for given taxon and site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsString" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsString"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsString</span></tt></a>(*args)</td>
<td>Returns string representation of diploid values returned by genotype() for given taxon and site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringArray" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringArray"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringArray</span></tt></a>(*args)</td>
<td>Returns string representation of diploid values returned by genotypeArray() for given taxon and site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRange" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRange"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRange</span></tt></a>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRow" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRow"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeAsStringRow</span></tt></a>(*args)</td>
<td>Returns string representation of diploid alleles for given taxon for all sites.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeMatrix" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeMatrix"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeMatrix</span></tt></a>(*args)</td>
<td>Returns the immutable genotype matrix.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeRange" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeRange"><tt class="xref py py-obj docutils literal"><span class="pre">genotypeRange</span></tt></a>(*args)</td>
<td>Returns sequence of diploid allele values for given taxon in specified range (end site excluded).</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">getArray</span></tt>(size)</td>
<td>Gets an empty wrapped java array that can accept the type of the wrapped</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">getDblArray</span></tt>(rows[,&nbsp;cols])</td>
<td>Gets an empty wrapped java array that can accept the type of other wrapped java arrays: i.e.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllSites" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllSites"><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllSites</span></tt></a>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at each</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllTaxa" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllTaxa"><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForAllTaxa</span></tt></a>(*args)</td>
<td>Returns a sequence of true/false values indicating whether site at each taxon for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForSitesBlock" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForSitesBlock"><tt class="xref py py-obj docutils literal"><span class="pre">haplotypeAllelePresenceForSitesBlock</span></tt></a>(*args)</td>
<td>Returns sequence of true/false values indicating whether taxon at sites (in given blocks, 64 sites per block including start block but excluding end block) for given parent matches a specific allele (based on frequency).</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasDepth" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasDepth"><tt class="xref py py-obj docutils literal"><span class="pre">hasDepth</span></tt></a>(*args)</td>
<td>Returns true if this genotype table has sequencing depth</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasReference" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasReference"><tt class="xref py py-obj docutils literal"><span class="pre">hasReference</span></tt></a>(*args)</td>
<td>Returns whether this genotype table has defined reference sequence.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasSiteScores" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasSiteScores"><tt class="xref py py-obj docutils literal"><span class="pre">hasSiteScores</span></tt></a>(*args)</td>
<td>Returns true if this genotype table has site scores</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hashCode</span></tt>(*args)</td>
<td>Returns a hash code vlaue for the object</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCount" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCount"><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCount</span></tt></a>(*args)</td>
<td>Returns number of heterozygous taxa at given site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCountForTaxon" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCountForTaxon"><tt class="xref py py-obj docutils literal"><span class="pre">heterozygousCountForTaxon</span></tt></a>(*args)</td>
<td>Returns number of heterozygous sites at given taxon</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.indelSize" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.indelSize"><tt class="xref py py-obj docutils literal"><span class="pre">indelSize</span></tt></a>(*args)</td>
<td>Returns size of indel at given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isAllPolymorphic" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isAllPolymorphic"><tt class="xref py py-obj docutils literal"><span class="pre">isAllPolymorphic</span></tt></a>(*args)</td>
<td>Returns whether all sites are polymorphic</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isHeterozygous" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isHeterozygous"><tt class="xref py py-obj docutils literal"><span class="pre">isHeterozygous</span></tt></a>(*args)</td>
<td>Returns whether allele values at given taxon and site are heterozygous.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isIndel" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isIndel"><tt class="xref py py-obj docutils literal"><span class="pre">isIndel</span></tt></a>(*args)</td>
<td>Returns whether given site is an indel</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPhased" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPhased"><tt class="xref py py-obj docutils literal"><span class="pre">isPhased</span></tt></a>(*args)</td>
<td>Returns whether this genotype table is phased</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPolymorphic" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPolymorphic"><tt class="xref py py-obj docutils literal"><span class="pre">isPolymorphic</span></tt></a>(*args)</td>
<td>Returns whether given site is polymorphic</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPositiveStrand" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPositiveStrand"><tt class="xref py py-obj docutils literal"><span class="pre">isPositiveStrand</span></tt></a>(*args)</td>
<td>Returns whether is positive strand at given site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAllele" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAllele"><tt class="xref py py-obj docutils literal"><span class="pre">majorAllele</span></tt></a>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleAsString" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleAsString"><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleAsString</span></tt></a>(*args)</td>
<td>Returns most common allele at given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleCount" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleCount"><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleCount</span></tt></a>(*args)</td>
<td>Returns the major allele count for given site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleFrequency" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleFrequency"><tt class="xref py py-obj docutils literal"><span class="pre">majorAlleleFrequency</span></tt></a>(*args)</td>
<td>Returns frequency for major allele at given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorMinorCounts" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorMinorCounts"><tt class="xref py py-obj docutils literal"><span class="pre">majorMinorCounts</span></tt></a>(*args)</td>
<td>Returns counts of all major/minor allele combinations from highest frequency to lowest for whole genotype table.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.maxNumAlleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.maxNumAlleles"><tt class="xref py py-obj docutils literal"><span class="pre">maxNumAlleles</span></tt></a>(*args)</td>
<td>Returns max number of alleles defined for any given site</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAllele" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAllele"><tt class="xref py py-obj docutils literal"><span class="pre">minorAllele</span></tt></a>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleAsString" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleAsString"><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleAsString</span></tt></a>(*args)</td>
<td>Returns most common minor allele at given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleCount" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleCount"><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleCount</span></tt></a>(*args)</td>
<td>Returns the minor allele count for given site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleFrequency" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleFrequency"><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleleFrequency</span></tt></a>(*args)</td>
<td>Returns frequency for most common minor allele at given site.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleles"><tt class="xref py py-obj docutils literal"><span class="pre">minorAlleles</span></tt></a>(*args)</td>
<td>Returns all minor alleles at given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numChromosomes" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numChromosomes"><tt class="xref py py-obj docutils literal"><span class="pre">numChromosomes</span></tt></a>(*args)</td>
<td>Return number of chromosomes</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfSites" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfSites"><tt class="xref py py-obj docutils literal"><span class="pre">numberOfSites</span></tt></a>(*args)</td>
<td>Returns total number of sites of this genotype table</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfTaxa" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfTaxa"><tt class="xref py py-obj docutils literal"><span class="pre">numberOfTaxa</span></tt></a>(*args)</td>
<td>Returns the number of taxa</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.physicalPositions" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.physicalPositions"><tt class="xref py py-obj docutils literal"><span class="pre">physicalPositions</span></tt></a>(*args)</td>
<td>Returns all physical positions</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.positions" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.positions"><tt class="xref py py-obj docutils literal"><span class="pre">positions</span></tt></a>(*args)</td>
<td>Return the position list for the gneotype table</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAllele" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAllele"><tt class="xref py py-obj docutils literal"><span class="pre">referenceAllele</span></tt></a>(*args)</td>
<td>Returns reference diploid allele values at given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleleForAllSites" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleleForAllSites"><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleleForAllSites</span></tt></a>(*args)</td>
<td>Returns reference sequence of diploid allele values.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleles"><tt class="xref py py-obj docutils literal"><span class="pre">referenceAlleles</span></tt></a>(*args)</td>
<td>Returns reference sequence of diploid allele values for given taxon in specified range (end site not included).</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.retainsRareAlleles" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.retainsRareAlleles"><tt class="xref py py-obj docutils literal"><span class="pre">retainsRareAlleles</span></tt></a>(*args)</td>
<td>Returns true if this genotype table retains rare alleles.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteName" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteName"><tt class="xref py py-obj docutils literal"><span class="pre">siteName</span></tt></a>(*args)</td>
<td>Get SNP ID for specified site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteOfPhysicalPosition" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteOfPhysicalPosition"><tt class="xref py py-obj docutils literal"><span class="pre">siteOfPhysicalPosition</span></tt></a>(*args)</td>
<td>Returns sites of given physical position/SNP ID in chromosome.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScore" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScore"><tt class="xref py py-obj docutils literal"><span class="pre">siteScore</span></tt></a>(*args)</td>
<td>Returns the site score of the given taxon and site</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScoreType" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScoreType"><tt class="xref py py-obj docutils literal"><span class="pre">siteScoreType</span></tt></a>(*args)</td>
<td>Return what type of site scores this genotype table has</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScores" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScores"><tt class="xref py py-obj docutils literal"><span class="pre">siteScores</span></tt></a>(*args)</td>
<td>Returns the site scores</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxa" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxa"><tt class="xref py py-obj docutils literal"><span class="pre">taxa</span></tt></a>(*args)</td>
<td>Return taxa list of this genotype table</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxaName" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxaName"><tt class="xref py py-obj docutils literal"><span class="pre">taxaName</span></tt></a>(*args)</td>
<td>Returns taxa name at given index</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">toString</span></tt>(*args)</td>
<td>Returns a string representation of the object</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForSite" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForSite"><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForSite</span></tt></a>(*args)</td>
<td>Returns total number of non-missing allele values for given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForTaxon" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForTaxon"><tt class="xref py py-obj docutils literal"><span class="pre">totalGametesNonMissingForTaxon</span></tt></a>(*args)</td>
<td>Returns total number of non-missing allele values for given taxon.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForSite" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForSite"><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForSite</span></tt></a>(*args)</td>
<td>Returns total number of non-missing taxa for given site.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForTaxon" title="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForTaxon"><tt class="xref py py-obj docutils literal"><span class="pre">totalNonMissingForTaxon</span></tt></a>(*args)</td>
<td>Returns total number of non-missing sites for given taxon.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">wrap_existing_array</span></tt>(arr_instance)</td>
<td>Wraps a java array of this class&#8217;s type</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.ALLELE_SORT_TYPE">
<tt class="descname">ALLELE_SORT_TYPE</tt><em class="property"> = &lt;Frequency: 0, Depth: 1, Global_Frequency: 2, Reference: 3&gt;</em><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.ALLELE_SORT_TYPE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.SITE_SCORE_TYPE">
<tt class="descname">SITE_SCORE_TYPE</tt><em class="property"> = &lt;None: 0, MixedScoreTypes: 1, QualityScore: 2, ImputedProbablity: 3, Dosage: 4&gt;</em><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.SITE_SCORE_TYPE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.__init__">
<tt class="descname">__init__</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.__init__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleleDefinitions">
<tt class="descname">alleleDefinitions</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.alleleDefinitions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleleDefinitions" title="Permalink to this definition">¶</a></dt>
<dd><p>For the first signature:
Returns allele values as strings for all sites. The first dimension
of this array indexes the sites. The second dimension indexes the allele
values for the given site. The indices for the allele values are used as
the codes to store data. These codes (indices) are returned by the genotype()
methods. If only one array of allele values is returned, that is the
encoding for all sites
For the second signature:
Same as alleleDefinitions() for only one site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>alleleDefinitions</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><em>alleleDefinitions</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">String[][] alleleDefinitions() allele values for all sites String[] alleleDefinitions(int site) allele values for given site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">String[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllSites">
<tt class="descname">allelePresenceForAllSites</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.allelePresenceForAllSites"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllSites" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of true/false values indicating whether taxon at
each site matches a specific allele (based on frequency). Allele
number of value 0 would be the major allele. Allele number of value 1
would be the minor allele. Allele number of value 2 would be the third
most frequent allele value and so on.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>allelePresenceForAllSites</em> (int <strong>taxon</strong>, WHICH_ALLELE <strong>allele</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele (WHICH_ALLELE)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Sequence of true/false value (BitSet)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">BitSet</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllTaxa">
<tt class="descname">allelePresenceForAllTaxa</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.allelePresenceForAllTaxa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForAllTaxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of true/false values indicating whether site at
each taxon matches a specific allele</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>allelePresenceForAllTaxa</em> (int <strong>site</strong>, WHICH_ALLELE <strong>allele</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">sequence of true/false values</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">BitSet</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForSitesBlock">
<tt class="descname">allelePresenceForSitesBlock</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.allelePresenceForSitesBlock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelePresenceForSitesBlock" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of true/false values indicating whether taxon at site
(in given blocks, 64 sites per block including start block but excluding
end block) matches a specific allele</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>allelePresenceForSitesBlock</em> (int <strong>taxon</strong>, WHICH_ALLELE <strong>allele</strong>, int <strong>startBlock</strong>, int <strong>endBlock</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele (WHICH_ALLELE)</li>
<li><strong>startBlock</strong> (<em>int</em>) &#8211; starting block (int)</li>
<li><strong>endBlock</strong> (<em>int</em>) &#8211; end block (int)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Sequence of true/false values (long[])</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">long[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleles">
<tt class="descname">alleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.alleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.alleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all alleles at given site in order of frequency. Gap
is included as state. Heterozygous count one for each allele value.
Homozygous counts two for the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>alleles</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">All alleles</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesBySortType">
<tt class="descname">allelesBySortType</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.allelesBySortType"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesBySortType" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all alleles at given site in order defined by scope</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>allelesBySortType</em> (GenotypeTable.ALLELE_SORT_TYPE <strong>scope</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>scope</strong> (<em>GenotypeTable.ALLELE_SORT_TYPE</em>) &#8211; scope</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">alleles</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">byte[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesSortedByFrequency">
<tt class="descname">allelesSortedByFrequency</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.allelesSortedByFrequency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.allelesSortedByFrequency" title="Permalink to this definition">¶</a></dt>
<dd><p>Return sorted list of alleles from highest frequency to lowest
at given site in genotype table. Resulting double dimension array
holds alleles (bytes) in result[0]. And the counts are in result[1].
Counts haploid values twice and diploid values once. Higher ploids
are not supported</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>allelesSortedByFrequency</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">sorted list of alleles and counts</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.bitStorage">
<tt class="descname">bitStorage</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.bitStorage"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.bitStorage" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns BitStorage for this Genotype</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>bitStorage</em> (GenotypeTable.WHICH_ALLELE <strong>allele</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>allele</strong> (<em>GenotypeTable.WHICH_ALLELE</em>) &#8211; allele</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">BitStorage</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">BitStorage</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomalPosition">
<tt class="descname">chromosomalPosition</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosomalPosition"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomalPosition" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the physical position at given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>chromosomalPosition</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">physical position</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosome">
<tt class="descname">chromosome</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosome" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns Chromosome for given site or with matching name
(first to match will be returned)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>chromosome</em> (int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>chromosome</em> (String <strong>name</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>site</strong> (<em>int</em>) &#8211; site Chromosome chromosome(String name):</li>
<li><strong>name</strong> (<em>String</em>) &#8211; name</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Chromosome</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">Chromosome</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeName">
<tt class="descname">chromosomeName</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosomeName"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeName" title="Permalink to this definition">¶</a></dt>
<dd><p>Return chromosome name for given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>chromosomeName</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Chromosome name</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeSiteCount">
<tt class="descname">chromosomeSiteCount</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosomeSiteCount"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomeSiteCount" title="Permalink to this definition">¶</a></dt>
<dd><p>Return number of sites for given chromosome</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>chromosomeSiteCount</em> (Chromosome <strong>chromosome</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>chromosome</strong> (<em>Chromosome</em>) &#8211; chromosome</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Number of sites</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomes">
<tt class="descname">chromosomes</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosomes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomes" title="Permalink to this definition">¶</a></dt>
<dd><p>Return all chromosomes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>chromosomes</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">chromosomes</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Chromosome[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomesOffsets">
<tt class="descname">chromosomesOffsets</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.chromosomesOffsets"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.chromosomesOffsets" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns starting site for each chromosome</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>chromosomesOffsets</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Starting site for each chromosome</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.compositeAlignments">
<tt class="descname">compositeAlignments</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.compositeAlignments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.compositeAlignments" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns individual genotype tables within this genotype table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>compositeAlignments</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of genotype tables</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">GenotypeTable[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depth">
<tt class="descname">depth</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.depth"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depth" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns allele depth object (null if not present)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>depth</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">allele depth associated with genotypeTable</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">AlleleDepth</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depthForAlleles">
<tt class="descname">depthForAlleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.depthForAlleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.depthForAlleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns depth count for each diploid allele at the given taxon and site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>depthForAlleles</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">two counts</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.diploidAsString">
<tt class="descname">diploidAsString</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.diploidAsString"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.diploidAsString" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns String representation of diploid allele value at site.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>diploidAsString</em> (int <strong>site</strong>, byte <strong>value</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
<li><strong>value</strong> (<em>byte</em>) &#8211; diploid allele value</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">String representation</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">String</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.firstLastSiteOfChromosome">
<tt class="descname">firstLastSiteOfChromosome</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.firstLastSiteOfChromosome"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.firstLastSiteOfChromosome" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the first (inclusive) and last (exclusive) site of the specified
chromosome in this genotype table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>firstLastSiteOfChromosome</em> (Chromosome <strong>chromosome</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>chromosome</strong> (<em>Chromosome</em>) &#8211; chromosome</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">first and last site</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genoCounts">
<tt class="descname">genoCounts</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genoCounts"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genoCounts" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns counts of all diploid combinations from highest frequency
to lowest for whole genotype table. Resulting double dimension array
holds diploids (Strings) in result[0]. And the counts are in result[1] (Longs)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genoCounts</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">diploid counts</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Object[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genomeVersion">
<tt class="descname">genomeVersion</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genomeVersion"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genomeVersion" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets the Genome Assembly</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genomeVersion</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">The genome assembly</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genosSortedByFrequency">
<tt class="descname">genosSortedByFrequency</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genosSortedByFrequency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genosSortedByFrequency" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sorted list of diploid values from highest frequency to
lowest at given site in genotype table. Resulting double dimension array
holds diploids (Strings) in result[0]. And the counts are in result[1]
(Integers).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genosSortedByFrequency</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Sorted list of diploids and counts</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">Object[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotype">
<tt class="descname">genotype</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotype"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotype" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns diploid value (genotype) for a given taxon and site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotype</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotype</em> (int <strong>taxon</strong>, Chromosome <strong>chromosome</strong>, int <strong>physicalPosition</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site genotype(int taxon, Chromosome chromosome, int physicalPosition):</li>
<li><strong>chromosome</strong> (<em>Chromosome</em>) &#8211; chromosome</li>
<li><strong>physicalPosition</strong> (<em>int</em>) &#8211; physicalPosition</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">High four bits generally encode more frequent allele and lower four bits encode the less frequent allele</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">byte</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllSites">
<tt class="descname">genotypeAllSites</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAllSites"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllSites" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of diploid allele values for all taxa for given site.
Each value in array is what would be returned by genotype().</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genotypeAllSites</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Sequence of diploid allele values</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllTaxa">
<tt class="descname">genotypeAllTaxa</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAllTaxa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAllTaxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of diploid allele values for all taxa for given
site. Each value in array is what would be returned by genotype().</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genotypeAllTaxa</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Sequence of diploid allele values</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeArray">
<tt class="descname">genotypeArray</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeArray"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeArray" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns diploid values for given taxon and site. Same values as
genotype(), except two values are already separated into two bytes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeArray</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">First byte (index 0) holds first allele value in right-most four bits. Second byte (index 1) holds second allele value in right-most four bits</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">byte[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsString">
<tt class="descname">genotypeAsString</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAsString"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsString" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns string representation of diploid values returned by genotype()
for given taxon and site. The two allele values will be separated by a
colon (:) delimiter.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeAsString</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeAsString</em> (int <strong>site</strong>, byte <strong>value</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site (int)</li>
<li><strong>value</strong> (<em>byte</em>) &#8211; allele value</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">string representation of diploid values</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">String</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringArray">
<tt class="descname">genotypeAsStringArray</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAsStringArray"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringArray" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns string representation of diploid values returned by genotypeArray() for
given taxon and site. Same two allele values as genotypeAsString(), except
already separated by two Strings.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeAsStringArray</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site (int)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">string representation of diploid values</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">String[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRange">
<tt class="descname">genotypeAsStringRange</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAsStringRange"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRange" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns string representation of diploid alleles for given taxon
in specified range (end site excluded). Each value in string is what would
be retuned by genotypeAsString()</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeAsStringRange</em> (int <strong>taxon</strong>, int <strong>startSite</strong>, int <strong>endSite</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>startSite</strong> (<em>int</em>) &#8211; start site (int)</li>
<li><strong>endSite</strong> (<em>int</em>) &#8211; end site (int)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">string representation of alleles in range</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">String</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRow">
<tt class="descname">genotypeAsStringRow</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeAsStringRow"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeAsStringRow" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns string representation of diploid alleles for given taxon for all
sites. Each value in string is what would be returned by genotypeAsString()</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genotypeAsStringRow</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">string representaiton of alleles</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeMatrix">
<tt class="descname">genotypeMatrix</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeMatrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeMatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the immutable genotype matrix. Taxa and positions are not
part of the matrix. This method is used for copying genotype tables,
when either the taxa or positions have changed</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>genotypeMatrix</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Genotype matrix</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">GenotypeCallTable</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeRange">
<tt class="descname">genotypeRange</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.genotypeRange"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.genotypeRange" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of diploid allele values for given taxon in specified
range (end site excluded). Each value in array is what would be returned
by genotype()</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>genotypeRange</em> (int <strong>taxon</strong>, int <strong>startSite</strong>, int <strong>endSite</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon</li>
<li><strong>startSite</strong> (<em>int</em>) &#8211; start site</li>
<li><strong>endSite</strong> (<em>int</em>) &#8211; end site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Sequence of diploid allele values</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">byte[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllSites">
<tt class="descname">haplotypeAllelePresenceForAllSites</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.haplotypeAllelePresenceForAllSites"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllSites" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of true/false values indicating whether taxon at each
site for given parent matches a specific allele</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>haplotypeAllelePresenceForAllSites</em> (int <strong>taxon</strong>, boolean <strong>firstParent</strong>, WHICH_ALLELE <strong>allele</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>firstParent</strong> (<em>boolean</em>) &#8211; true for first parent (false for second parent)</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele (WHICH_ALLELE)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Sequence of true/false values (BitSet)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">BitSet</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllTaxa">
<tt class="descname">haplotypeAllelePresenceForAllTaxa</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.haplotypeAllelePresenceForAllTaxa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForAllTaxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a sequence of true/false values indicating whether site at each
taxon for given parent matches a specific allele (based on frequency).
Allele number of value 0 would be the major allele. Allele number of value
1 would be the minor allele. Allele number of value 2 would be the third
most frequent allele value and so on</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>haplotypeAllelePresenceForAllTaxa</em> (int <strong>site</strong>, boolean <strong>firstParent</strong>, WHICH_ALLELE <strong>allele</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>site</strong> (<em>int</em>) &#8211; site (int)</li>
<li><strong>firstParent</strong> (<em>boolean</em>) &#8211; true for first parent (false for second parent)</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele (WHICH_ALLELE)</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Sequence of true/false values (BitSet)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">BitSet</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForSitesBlock">
<tt class="descname">haplotypeAllelePresenceForSitesBlock</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.haplotypeAllelePresenceForSitesBlock"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.haplotypeAllelePresenceForSitesBlock" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sequence of true/false values indicating whether taxon at sites
(in given blocks, 64 sites per block including start block but excluding
end block) for given parent matches a specific allele (based on frequency).
Allele number of value 0 would be the major allele. Allele number of value 1
would be the minor allele. Allele number of value 2 would be the third most
frequent allele value and so on.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>haplotypeAllelePresenceForSitesBlock</em> (int <strong>taxon</strong>, boolean <strong>firstParent</strong>, WHICH_ALLELE <strong>allele</strong>, int <strong>startBlock</strong>, int <strong>endBlock</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon (int)</li>
<li><strong>firstParent</strong> (<em>boolean</em>) &#8211; true for first parent (false for second parent)</li>
<li><strong>allele</strong> (<em>WHICH_ALLELE</em>) &#8211; allele (WHICH_ALLELE)</li>
<li><strong>startBlock</strong> (<em>int</em>) &#8211; starting block (int)</li>
<li><strong>endBlock</strong> (<em>int</em>) &#8211; ending block (int)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasDepth">
<tt class="descname">hasDepth</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.hasDepth"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasDepth" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns true if this genotype table has sequencing depth</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>hasDepth</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if this genotype table has sequencing depth</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasReference">
<tt class="descname">hasReference</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.hasReference"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasReference" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether this genotype table has defined reference sequence.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>hasReference</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if this genotype table has reference sequence</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasSiteScores">
<tt class="descname">hasSiteScores</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.hasSiteScores"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.hasSiteScores" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns true if this genotype table has site scores</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>hasSiteScores</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">True if this genotype table has site scores</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCount">
<tt class="descname">heterozygousCount</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.heterozygousCount"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCount" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns number of heterozygous taxa at given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>heterozygousCount</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Number of heterozygous taxa</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCountForTaxon">
<tt class="descname">heterozygousCountForTaxon</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.heterozygousCountForTaxon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.heterozygousCountForTaxon" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns number of heterozygous sites at given taxon</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>heterozygousCountForTaxon</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>taxon</strong> (<em>int</em>) &#8211; taxon</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">number of heterozygous sites</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.indelSize">
<tt class="descname">indelSize</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.indelSize"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.indelSize" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns size of indel at given site.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>indelSize</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">indel size</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isAllPolymorphic">
<tt class="descname">isAllPolymorphic</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isAllPolymorphic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isAllPolymorphic" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether all sites are polymorphic</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isAllPolymorphic</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">true if all sites are polymorphic</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isHeterozygous">
<tt class="descname">isHeterozygous</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isHeterozygous"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isHeterozygous" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether allele values at given taxon and site are heterozygous.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>isHeterozygous</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Whether heterozygous</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">boolean</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isIndel">
<tt class="descname">isIndel</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isIndel"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isIndel" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether given site is an indel</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isIndel</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">true if indel</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPhased">
<tt class="descname">isPhased</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isPhased"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPhased" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether this genotype table is phased</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isPhased</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">true if phased</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPolymorphic">
<tt class="descname">isPolymorphic</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isPolymorphic"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPolymorphic" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether given site is polymorphic</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isPolymorphic</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">true if given site is polymorphic</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPositiveStrand">
<tt class="descname">isPositiveStrand</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.isPositiveStrand"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.isPositiveStrand" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns whether is positive strand at given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>isPositiveStrand</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">whether is positive strand</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAllele">
<tt class="descname">majorAllele</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.majorAllele"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAllele" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns most common allele at given site. Gap is included as state.
Heterozygous count one for each allele value. Homozygous counts two for
the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>majorAllele</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">most common allele</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleAsString">
<tt class="descname">majorAlleleAsString</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.majorAlleleAsString"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleAsString" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns most common allele at given site. Gap is included as state.
Heterozygous count one for each allele value. Homozygous counts two for
allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>majorAlleleAsString</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">most common allele as String</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleCount">
<tt class="descname">majorAlleleCount</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.majorAlleleCount"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleCount" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the major allele count for given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>majorAlleleCount</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">major allele count</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleFrequency">
<tt class="descname">majorAlleleFrequency</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.majorAlleleFrequency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorAlleleFrequency" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns frequency for major allele at given site. Gap is included as
state. Heterozygous count one for each allele value. Homozygous counts
two for the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>majorAlleleFrequency</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">frequency</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">double</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorMinorCounts">
<tt class="descname">majorMinorCounts</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.majorMinorCounts"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.majorMinorCounts" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns counts of all major/minor allele combinations from highest
frequency to lowest for whole genotype table. Resulting double dimension
array holds major/minor allele (Strings) in result[0]. And the counts are
in result[1] (Longs)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>majorMinorCounts</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">diploid counts</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">Object[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.maxNumAlleles">
<tt class="descname">maxNumAlleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.maxNumAlleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.maxNumAlleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns max number of alleles defined for any given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>maxNumAlleles</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">max number of alleles</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAllele">
<tt class="descname">minorAllele</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.minorAllele"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAllele" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns most common minor allele at given site. Gap is included as state.
Heterozygous count one for each allele value. Homozygous counts two for
the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>minorAllele</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">most common minor allele</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleAsString">
<tt class="descname">minorAlleleAsString</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.minorAlleleAsString"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleAsString" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns most common minor allele at given site. Gap is included as state.
Heterozygous count one for each allele value. Homozygous counts two for
allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>minorAlleleAsString</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">most common minor allele as String</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleCount">
<tt class="descname">minorAlleleCount</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.minorAlleleCount"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleCount" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the minor allele count for given site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>minorAlleleCount</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">minor allele count</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleFrequency">
<tt class="descname">minorAlleleFrequency</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.minorAlleleFrequency"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleleFrequency" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns frequency for most common minor allele at given site.
Gap is included as state. Heterozygous count one for each allele value.
Homozygous counts two for the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>minorAlleleFrequency</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">frequency</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">double</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleles">
<tt class="descname">minorAlleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.minorAlleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.minorAlleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all minor alleles at given site. Gap is included as state
Heterozygous count one for each allele value. Homozygous counts two
for the allele value</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>minorAlleles</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numChromosomes">
<tt class="descname">numChromosomes</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.numChromosomes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numChromosomes" title="Permalink to this definition">¶</a></dt>
<dd><p>Return number of chromosomes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>numChromosomes</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Number of chromosomes</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfSites">
<tt class="descname">numberOfSites</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.numberOfSites"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfSites" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns total number of sites of this genotype table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>numberOfSites</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Number of sites</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfTaxa">
<tt class="descname">numberOfTaxa</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.numberOfTaxa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.numberOfTaxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the number of taxa</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>numberOfTaxa</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Number of taxa</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.physicalPositions">
<tt class="descname">physicalPositions</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.physicalPositions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.physicalPositions" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns all physical positions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>physicalPositions</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">physical positions</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">int[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.positions">
<tt class="descname">positions</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.positions"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.positions" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the position list for the gneotype table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>positions</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">PositionList for all sites</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">PositionList</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAllele">
<tt class="descname">referenceAllele</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.referenceAllele"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAllele" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns reference diploid allele values at given site.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>referenceAllele</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">First 4 bits are the first allele value and the second 4 bits are the second allele value.</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">byte</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleleForAllSites">
<tt class="descname">referenceAlleleForAllSites</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.referenceAlleleForAllSites"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleleForAllSites" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns reference sequence of diploid allele values. Each value
in array contains both diploid values. First 4 bits holds the first allele,
and the second 4 bits holds the second allele</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>referenceAlleleForAllSites</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Reference sequence of diploid allele values</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">byte[]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleles">
<tt class="descname">referenceAlleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.referenceAlleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.referenceAlleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns reference sequence of diploid allele values for given taxon in specified
range (end site not included). Each value in array contains both diploid values. First 4
bits holds the first allele, and the second 4 bits holds the second allele</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>referenceAlleles</em> (int <strong>startSite</strong>, int <strong>endSite</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>startSite</strong> (<em>int</em>) &#8211; start site</li>
<li><strong>endSite</strong> (<em>int</em>) &#8211; end site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Reference sequence of diploid allele values</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">byte[]</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.retainsRareAlleles">
<tt class="descname">retainsRareAlleles</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.retainsRareAlleles"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.retainsRareAlleles" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns true if this genotype table retains rare alleles.
If false, rare alleles are recorded as unknown</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>retainsRareAlleles</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Whether rare alleles are retained</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">boolean</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteName">
<tt class="descname">siteName</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.siteName"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteName" title="Permalink to this definition">¶</a></dt>
<dd><p>Get SNP ID for specified site</p>
<p>Signature:</p>
<p>String siteName(int site)</p>
<p>Arguments:</p>
<p>site &#8211; site</p>
<p>Returns:</p>
<p>Site name</p>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteOfPhysicalPosition">
<tt class="descname">siteOfPhysicalPosition</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.siteOfPhysicalPosition"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteOfPhysicalPosition" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns sites of given physical position/SNP ID in chromosome. If the
physical position doesn&#8217;t exist, (-(insertion point) -1) is returned. If chromosome
is not found, an exception is thrown. This can support multiple sites with the same
physical position but different SNP IDs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>siteOfPhysicalPosition</em> (int <strong>physicalPosition</strong>, Chromosome <strong>chromosome</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>siteOfPhysicalPosition</em> (int <strong>physicalPosition</strong>, Chromosome <strong>chromosome</strong>, String <strong>snpName</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>physicalPosition</strong> (<em>int</em>) &#8211; physical position</li>
<li><strong>chromosome</strong> (<em>Chromosome</em>) &#8211; chromosome. If null, the first chromosome is used</li>
<li><strong>snpName</strong> (<em>String</em>) &#8211; SNP ID</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Index</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">int</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScore">
<tt class="descname">siteScore</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.siteScore"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScore" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the site score of the given taxon and site</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><p class="first"><em>siteScore</em> (int <strong>taxon</strong>, int <strong>site</strong>)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>taxon</strong> (<em>int</em>) &#8211; taxon index</li>
<li><strong>site</strong> (<em>int</em>) &#8211; site</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">site score</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body"><p class="first last">float</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScoreType">
<tt class="descname">siteScoreType</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.siteScoreType"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScoreType" title="Permalink to this definition">¶</a></dt>
<dd><p>Return what type of site scores this genotype table has</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>siteScoreType</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Site score type</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">GenotypeTable.SITE_SCORE_TYPE</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScores">
<tt class="descname">siteScores</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.siteScores"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.siteScores" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the site scores</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>siteScores</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">site scores</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">float[][]</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxa">
<tt class="descname">taxa</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.taxa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxa" title="Permalink to this definition">¶</a></dt>
<dd><p>Return taxa list of this genotype table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>taxa</em> ()</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">taxa list</td>
</tr>
<tr class="field-odd field"><th class="field-name">Return type:</th><td class="field-body">TaxaList</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxaName">
<tt class="descname">taxaName</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.taxaName"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.taxaName" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns taxa name at given index</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>taxaName</em> (int <strong>index</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>index</strong> (<em>int</em>) &#8211; index</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">taxa name</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">String</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForSite">
<tt class="descname">totalGametesNonMissingForSite</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.totalGametesNonMissingForSite"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForSite" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns total number of non-missing allele values for given site. This can
be twice the number of taxa, as diploid values are supported</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>totalGametesNonMissingForSite</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Number of non-missing allele values</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForTaxon">
<tt class="descname">totalGametesNonMissingForTaxon</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.totalGametesNonMissingForTaxon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalGametesNonMissingForTaxon" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns total number of non-missing allele values for given taxon. This
can be twice the number of sites, as diploid values are supported</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>totalGametesNonMissingForTaxon</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>taxon</strong> (<em>int</em>) &#8211; taxon</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Number of non-missing allele values</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForSite">
<tt class="descname">totalNonMissingForSite</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.totalNonMissingForSite"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForSite" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns total number of non-missing taxa for given site. Taxa are
considered missing only if both allele values are Unknown (N).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>totalNonMissingForSite</em> (int <strong>site</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>site</strong> (<em>int</em>) &#8211; site</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Number of non-missing taxa</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForTaxon">
<tt class="descname">totalNonMissingForTaxon</tt><big>(</big><em>*args</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#GenotypeTable.totalNonMissingForTaxon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.GenotypeTable.totalNonMissingForTaxon" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns total number of non-missing sites for given taxon. Sites
are considered missing only if both allele values are Unknown (N).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Signature:</th><td class="field-body"><em>totalNonMissingForTaxon</em> (int <strong>taxon</strong>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><strong>taxon</strong> (<em>int</em>) &#8211; taxon</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">number of non-missing sites</td>
</tr>
<tr class="field-even field"><th class="field-name">Return type:</th><td class="field-body">int</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.MetaGenotypeTable">
<em class="property">class </em><tt class="descclassname">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.</tt><tt class="descname">MetaGenotypeTable</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/GenotypeTable.html#MetaGenotypeTable"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.GenotypeTable.MetaGenotypeTable" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="TASSELpy.java.lang.html#TASSELpy.java.lang.Object.Object" title="TASSELpy.java.lang.Object.Object"><tt class="xref py py-class docutils literal"><span class="pre">TASSELpy.java.lang.Object.Object</span></tt></a></p>
<p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">clone</span></tt>(*args)</td>
<td>Creates and returns a copy of this object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">equals</span></tt>(*args)</td>
<td>Indicates whether some other object is &#8220;equal to&#8221; this one</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">getArray</span></tt>(size)</td>
<td>Gets an empty wrapped java array that can accept the type of the wrapped</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">getDblArray</span></tt>(rows[,&nbsp;cols])</td>
<td>Gets an empty wrapped java array that can accept the type of other wrapped java arrays: i.e.</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">hashCode</span></tt>(*args)</td>
<td>Returns a hash code vlaue for the object</td>
</tr>
<tr class="row-even"><td><tt class="xref py py-obj docutils literal"><span class="pre">toString</span></tt>(*args)</td>
<td>Returns a string representation of the object</td>
</tr>
<tr class="row-odd"><td><tt class="xref py py-obj docutils literal"><span class="pre">wrap_existing_array</span></tt>(arr_instance)</td>
<td>Wraps a java array of this class&#8217;s type</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-TASSELpy.net.maizegenetics.dna.snp.ImportUtils">
<span id="tasselpy-net-maizegenetics-dna-snp-importutils-module"></span><h2>TASSELpy.net.maizegenetics.dna.snp.ImportUtils module<a class="headerlink" href="#module-TASSELpy.net.maizegenetics.dna.snp.ImportUtils" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils">
<em class="property">class </em><tt class="descclassname">TASSELpy.net.maizegenetics.dna.snp.ImportUtils.</tt><tt class="descname">ImportUtils</tt><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/ImportUtils.html#ImportUtils"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils" title="Permalink to this definition">¶</a></dt>
<dd><p class="rubric">Methods</p>
<table border="1" class="longtable docutils">
<colgroup>
<col width="10%" />
<col width="90%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td><a class="reference internal" href="#TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils.readGuessFormat" title="TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils.readGuessFormat"><tt class="xref py py-obj docutils literal"><span class="pre">readGuessFormat</span></tt></a>(fileName)</td>
<td>Reads in a file after guessing its format</td>
</tr>
</tbody>
</table>
<dl class="staticmethod">
<dt id="TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils.readGuessFormat">
<em class="property">static </em><tt class="descname">readGuessFormat</tt><big>(</big><em>fileName</em><big>)</big><a class="reference internal" href="_modules/TASSELpy/net/maizegenetics/dna/snp/ImportUtils.html#ImportUtils.readGuessFormat"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#TASSELpy.net.maizegenetics.dna.snp.ImportUtils.ImportUtils.readGuessFormat" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads in a file after guessing its format</p>
<p>Arguments:</p>
<p>fileName &#8211; The name of the file</p>
<p>Returns:</p>
<p>A GenotypeTable</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-TASSELpy.net.maizegenetics.dna.snp">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-TASSELpy.net.maizegenetics.dna.snp" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">TASSELpy.net.maizegenetics.dna.snp package</a><ul>
<li><a class="reference internal" href="#subpackages">Subpackages</a></li>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#module-TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable">TASSELpy.net.maizegenetics.dna.snp.CoreGenotypeTable module</a></li>
<li><a class="reference internal" href="#module-TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable">TASSELpy.net.maizegenetics.dna.snp.FilterGenotypeTable module</a></li>
<li><a class="reference internal" href="#module-TASSELpy.net.maizegenetics.dna.snp.GenotypeTable">TASSELpy.net.maizegenetics.dna.snp.GenotypeTable module</a><ul>
</ul>
</li>
<li><a class="reference internal" href="#module-TASSELpy.net.maizegenetics.dna.snp.ImportUtils">TASSELpy.net.maizegenetics.dna.snp.ImportUtils module</a></li>
<li><a class="reference internal" href="#module-TASSELpy.net.maizegenetics.dna.snp">Module contents</a></li>
</ul>
</li>
</ul>

  <h4>Next topic</h4>
  <p class="topless"><a href="TASSELpy.net.maizegenetics.dna.snp.depth.html"
                        title="next chapter">TASSELpy.net.maizegenetics.dna.snp.depth package</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/TASSELpy.net.maizegenetics.dna.snp.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="TASSELpy.net.maizegenetics.dna.snp.depth.html" title="TASSELpy.net.maizegenetics.dna.snp.depth package"
             >next</a> |</li>
        <li><a href="index.html">TASSELpy 0.1.dev-20140723133829 documentation</a> &raquo;</li>
          <li><a href="TASSELpy.html" >TASSELpy package</a> &raquo;</li>
          <li><a href="TASSELpy.net.html" >TASSELpy.net package</a> &raquo;</li>
          <li><a href="TASSELpy.net.maizegenetics.html" >TASSELpy.net.maizegenetics package</a> &raquo;</li>
          <li><a href="TASSELpy.net.maizegenetics.dna.html" >TASSELpy.net.maizegenetics.dna package</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2014, Eli Rodgers-Melnick.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>